### DISCUSSION TITLE
How do you make the select field translatable?

### DISCUSSION THREAD

**Discussion by ghost** [area17/twill#1197] [Q&A] [answered]
I need to have a select field be translatable/vary by language in order to store the format of a video. I've read that this is possible and that I'd need to create a new select field or extend the current one, but there's no documentation on how to do so.

Any ideas?

**Comment by pboivin** [comment, marked as answer]
Hey @aarontilt , I think the most simple way at the moment would be to have one field per language with shared options. You can hook into the form's Vue.js layer to only show the field for the current language:
```blade
    @php
        $format_options = [
            ['value' => 1, 'label' => 'One'],
            ['value' => 2, 'label' => 'Two'],
            ['value' => 3, 'label' => 'Three'],
        ];
    @endphp

    <div v-show="$root.currentLocale.value === 'en'">
        @formField('select', [
            'name' => 'format_en',
            'label' => 'Format (EN)',
            'options' => $format_options,
        ])
    </div>

    <div v-show="$root.currentLocale.value === 'fr'">
        @formField('select', [
            'name' => 'format_fr',
            'label' => 'Format (FR)',
            'options' => $format_options,
        ])
    </div>
```

**Reply by pboivin** [reply]
Also, here's a bit of documentation on how to approach building/extending Twill components and form fields:
https://gist.github.com/pboivin/19d49cbb3c0a6c5c06469117597cc0f7

**Reply by ghost** [reply]
Aha! Thank you

**Comment by ifox** [comment]
Hi there, thanks @pboivin that's a nice way of doing this. 

~~Stepping back a little bit allow me to question the need for a translatable select. Since the values are either referencing IDs of another record (which itself would be translated) or static values, I don't really understand the need.~~

Scratch that, as I was writing it occured to me that you're looking to select a different value per language, which totally makes sense.
